classes/Search.php

```diff
@@ -229,22 +240,36 @@ public static function find(
             return false;
         }
 
+        $eligible_products2 = array();
         $intersect_array = array();
         $score_array = array();
+        $fuzzyLoop = 0;
         $words = Search::extractKeyWords($expr, $id_lang, false, $context->language->iso_code);
 
         foreach ($words as $key => $word) {
             if (!empty($word) && strlen($word) >= (int) Configuration::get('PS_SEARCH_MINWORDLEN')) {
                 $sql_param_search = self::getSearchParamFromWord($word);
-
-                $intersect_array[] = 'SELECT DISTINCT si.id_product
-					FROM ' . _DB_PREFIX_ . 'search_word sw
-					LEFT JOIN ' . _DB_PREFIX_ . 'search_index si ON sw.id_word = si.id_word
-					WHERE sw.id_lang = ' . (int) $id_lang . '
-						AND sw.id_shop = ' . $context->shop->id . '
-						AND sw.word LIKE
-					\'' . $sql_param_search . '\'';
-
+                $sql = 'SELECT DISTINCT si.id_product
+                            FROM ' . _DB_PREFIX_ . 'search_word sw
+                            LEFT JOIN ' . _DB_PREFIX_ . 'search_index si ON sw.id_word = si.id_word
+                            WHERE sw.id_lang = ' . (int) $id_lang . '
+                                AND sw.id_shop = ' . $context->shop->id . '
+                                AND sw.word LIKE';
+
+                while (!($result = $db->executeS($sql . "'" . $sql_param_search . "';", true, false))) {
+                    if (!Configuration::get('PS_SEARCH_FUZZY') ||
+                        $fuzzyLoop++ > Configuration::get('PS_SEARCH_FUZZY_MAX_LOOP') ||
```

Comment:
Can we use a variable defined before loop for all `Configuration::get` statements?

Reply:
Done